easybuild/tools/filetools.py

```diff
@@ -760,6 +760,17 @@ def verify_checksum(path, checksums):
         checksums = [checksums]
 
     for checksum in checksums:
+        if isinstance(checksum, dict):
+            filename = os.path.split(path)[-1]
+            if filename in checksum:
+                checksum = checksum[filename]
+            else:
+                # No checksum for this file
+                if build_option('enforce_checksums'):
+                    raise EasyBuildError("Missing checksum for %s", os.path.basename(path))
+                else:
+                    continue
```

Comment:
This can be trimmed:

```python
if filename in checksum:
    checksum = checksum[filename]
elif build_option('enforce_checksums'):
    raise EasyBuildError("Missing checksum for %s", filename)
else:
    continue
```

I'm not sure the `continue` part makes sense here though...

We should probably just set `checksum` to `None`, and let it fail elsewhere?